client/state/site-address-change/actions.js

```diff
@@ -126,7 +127,44 @@ export const requestSiteAddressChange = (
 		discard,
 	};
 
-	const errorHandler = ( error ) => {
+	dispatch( recordTracksEvent( 'calypso_siteaddresschange_request', eventProperties ) );
+
+	try {
+		const nonce = await fetchNonce( siteId );
+		const data = await wpcom
+			.undocumented()
+			.updateSiteAddress( siteId, newBlogName, domain, oldDomain, siteType, discard, nonce );
+
+		const newSlug = get( data, 'new_slug' );
+
+		if ( newSlug ) {
+			dispatch( recordTracksEvent( 'calypso_siteaddresschange_success', eventProperties ) );
+
+			await dispatch( requestSite( siteId ) );
+			// Re-fetch domains, as we changed the primary domain name
+			await dispatch( fetchSiteDomains( siteId ) );
+
+			dispatch( {
+				type: SITE_ADDRESS_CHANGE_REQUEST_SUCCESS,
+				siteId,
+			} );
+
+			dispatch(
+				successNotice( translate( 'Your new site address is ready to go!' ), {
+					id: 'siteAddressChangeSuccessful',
+					duration: 5000,
+					showDismiss: true,
+					isPersistent: true,
+				} )
+			);
+
+			// site slug, potentially freshly updated by the `requestSite` above
+			const siteSlug = getSiteSlug( getState(), siteId );
```

Comment:
Nice catch 👍  I wonder if we introduced this during the reduxification or it was a pre-existing bug.

Reply:
I believe we broke this years ago when reduxifying `sites-list`. For a Simple site with a mapped domain, its "canonical" slug is `example.blog`, the mapped domain. The unmapped `example.wordpress.com` is an alternative slug. The Flux `sites-list`, when doing the "get site ID by slug" operation, supported both of them. But the Redux selector always supported only the `example.blog` one.

#53328 is actually a part of effort to finally fix this 🙂